harbor/datadog_checks/harbor/harbor.py

```diff
@@ -108,18 +99,16 @@ def _submit_disk_metrics(self, api, base_tags):
 
     def check(self, instance):
         harbor_url = instance["url"]
-        username = instance["username"]
-        password = instance['password']
         tags = instance.get("tags", [])
         try:
             api = HarborAPI(harbor_url, self.http)
-            api.authenticate(username, password)
+            self._check_health(api, tags)
+            self._check_registries_health(api, tags)
+            self._submit_project_metrics(api, tags)
+            self._submit_disk_metrics(api, tags)
         except Exception:
             self.log.exception("Harbor API is not reachable")
```

Comment:
TODO: Update log message
The service check name doesn't really fit now (because it reports CRITICAL when maybe something else fails). But we don't have the concept of "connection" anymore as there is no session.